internal/bminventory/inventory.go

```diff
@@ -2549,16 +2549,22 @@ func (b *bareMetalInventory) PostStepReply(ctx context.Context, params installer
 	return installer.NewPostStepReplyNoContent()
 }
 
-func (b *bareMetalInventory) handleReplyError(params installer.PostStepReplyParams, ctx context.Context, log logrus.FieldLogger, h *models.Host) error {
-
-	if params.Reply.StepType == models.StepTypeInstall {
+func (b *bareMetalInventory) handleReplyError(params installer.PostStepReplyParams, ctx context.Context, log logrus.FieldLogger, h *models.Host, exitCode int64) error {
+	switch params.Reply.StepType {
+	case models.StepTypeInstall:
 		// Handle case of installation error due to an already running assisted-installer.
 		if params.Reply.ExitCode == ContainerAlreadyRunningExitCode && strings.Contains(params.Reply.Error, "the container name \"assisted-installer\" is already in use") {
 			log.Warnf("Install command failed due to an already running installation: %s", params.Reply.Error)
 			return nil
 		}
 		//if it's install step - need to move host to error
 		return b.hostApi.HandleInstallationFailure(ctx, h)
+	case models.StepTypeInstallationDiskSpeedCheck:
+		stepReply, err := filterReplyByType(params)
+		if err != nil {
+			return err
+		}
+		return b.processDiskSpeedCheckResponse(ctx, h, stepReply, exitCode)
```

Comment:
Why is this the only tool to be processed for an error? Isn't an error is when the tool wasn't able to run correctly?

Reply:
@ronniel1 said that if the tool failed once, probably it will fail again.  Therefore, he asked to add results in case of errors too.